apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -286,3 +292,23 @@ function wpcom_display_global_styles_banner_extra_tooltip() {
 	<?php
 }
 add_action( 'launch_bar_extra_tooltip_toggle_global_styles', 'wpcom_display_global_styles_banner_extra_tooltip' );
+
+/**
+ * Invalidates the cached Global Styles after changing them or when previewing them.
+ */
+function wpcom_invalidate_global_styles_cache() {
+	$transient_name = 'gutenberg_global_styles_' . get_stylesheet();
+
+	if ( wpcom_should_limit_global_styles() && wpcom_is_previewing_global_styles() ) {
+		add_filter( 'pre_transient_' . $transient_name, '__return_null' );
+		add_filter( 'pre_set_transient_' . $transient_name, '__return_false' );
```

Comment:
> Maybe we can request a filter called `gutenberg_global_styles_can_use_cached` that would serve as a discriminator for when we want to programmatically turn off the cache.

I totally agree. In fact, there is already a PR implementing this (https://github.com/WordPress/wordpress-develop/pull/2732, https://github.com/WordPress/gutenberg/pull/41201) but it looks like it didn't get too much attention.

Reply:
Maybe we can try to give that PR a second chance? We can always come back and merge this PR if necessary.

Reply:
Sure. But even if those PRs go forward, we'll still need to fix the cache problems in the meantime since it can take weeks until the changesets are available in WP.com (which is why we ~~temporary~~ disabled the default cache in Simple sites a few months ago: D81197-code).

@tyxla: do we have any way to expedite your PRs above?

Reply:
I don't, really. They have been lingering there for a while now. I wonder if we can get help from someone from .org. @annezazu do you have any idea who we can ask for help with reviews and shipping?

Reply:
On it :) I'll tag some folks in that PR to see if we can get things moving, considering 6.1 is now out the door.